helix-core/src/test/java/org/apache/helix/manager/zk/TestZKUtil.java

```diff
@@ -63,6 +65,13 @@ public void afterClass() {
     deleteCluster(clusterName);
   }
 
+  @AfterMethod()
+  public void afterMethod() {
+    String path = PropertyPathBuilder.instanceConfig(clusterName);
+    _gZkClient.deleteRecursively(path);
+    _gZkClient.createPersistent(path);
```

Comment:
- This is like delete and re-create the cluster, which I don't believe is a better way than deleting the path in the required methods.
- Not every method needs deleting. Ex. `testIsClusterSetup`.
- In my opinion, if a method creates the path, it is the method's responsibility to clean up.

Reply:
I agree it can cause unnecessary delete operation calls such as in `testIsClusterSetup`, but it doesn't add too much latency since what it does is deleting the nodes, not recreating the cluster . It is also safer than leaving the clean up responsibility to each method, as it can make sure the cluster is clean for each test method. I think both way work and it's barely a trade off.

Reply:
I checked the test cases one by one, there is no test which is testing to create the root path (please correct me if I am wrong). Given so, re-create the root path is easier and safer.
If we really have a case that requires the root path non-exists, feel free to change that test.